### DISCUSSION TITLE
Allow YAML config file (.renovaterc.yml, .renovaterc.yaml)

### DISCUSSION THREAD

**Discussion by timdp** [renovatebot/renovate#22410] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

In #969 and #1142, there was talk of supporting YAML for `.renovaterc`. The idea was abandoned in favor of JSON only. However:

* Many other tools support YAML config files. In the npm ecosystem in particular, [cosmiconfig](https://www.npmjs.com/package/cosmiconfig) is a popular choice.
* The later addition of JSON5 support proves that there is a need for a language that's more suited for composition and interpretation by humans.
* While there are some quirks to YAML, its broad adoption proves that these are manageable.
* YAML is well-understood by folks who typically integrate Renovate into their projects.
* YAML is a superset of JSON and the already published JSON Schema can be leveraged as-is.

I therefore propose that `.renovaterc.yml` and `.renovaterc.yaml` be added as supported configuration files, with a schema identical to that of JSON.

Note: I did search for existing discussions about this. While YAML came up a bunch of times, I couldn't find anything about the config file. Maybe I'm clumsy.

**Comment by viceice** [comment]
i think there is an issue already for it, but maybe closed because we abandoned that.

**Reply by viceice** [reply]
- #7031

**Reply by timdp** [reply]
Ah. I only searched the discussions because I'd started to create an issue and then arrived here. My bad.

I don't agree with https://github.com/renovatebot/renovate/issues/7031#issuecomment-676471156 for the reasons raised in my original post, but well, that's just my opinion.

Feel free to close either this discussion or the issue.
